crates/pop-cli/src/common/contracts.rs

```diff
@@ -21,19 +24,45 @@ pub async fn check_contracts_node_and_prompt() -> anyhow::Result<Option<PathBuf>
 			let spinner = spinner();
 			spinner.start("Sourcing substrate-contracts-node...");
 
-			let cache_path = crate::cache()?;
-			let binary = download_contracts_node(cache_path.clone()).await?;
+			binary.source(false, &(), true).await?;
 
 			spinner.stop(format!(
 				"substrate-contracts-node successfully sourced. Cached at: {}",
 				binary.path().to_str().unwrap()
 			));
-			node_path = Some(binary.path());
+			node_path = binary.path();
+		}
+	}
+	if binary.stale() {
```

Comment:
not for this PR, but I imagine this functionality is duplicated across parachains code?

Reply:
The function `stale()` is already in the `pop_common` crate

Reply:
sorry, should have been more clear. In terms of the entire section of "if stale do this, else do this"

Reply:
I think the logic that follows is quite specific, but I agree with you, it can be reviewed to see how we can refactor more of it into `pop_common`.